### DISCUSSION TITLE
Complete event never fires when using withFilter & graphql-redis-subscriptions

### DISCUSSION THREAD

**Discussion by ajmerik121** [enisdenjo/graphql-ws#265] [Q&A] [answered]
**Server code example**

![Screenshot_13](https://user-images.githubusercontent.com/74120006/141155600-d316dade-93d7-424c-a696-e670f6535e81.png)

**Client side code example**
![Screenshot_14](https://user-images.githubusercontent.com/74120006/141155808-287ece47-3456-48b5-8fdb-c323697b0d8a.png)

**Issue**
![pingggggggggggg](https://user-images.githubusercontent.com/74120006/141156686-93cb43d9-63c4-4d98-bca2-9b9654f87f28.png)

**Dependecies**
`
    "apollo-server-express": "^3.5.0",
    "graphql": "^16.0.1",
    "graphql-redis-subscriptions": "^2.4.0",
    "graphql-subscriptions": "^2.0.0",
    "graphql-ws": "^5.5.5",
    "ioredis": "^4.28.0",
`

**Comment by enisdenjo** [comment, marked as answer]
`graphql-ws` has extensive tests around completion of subscription iterables. This is very likely a problem with one of the other dependencies not completing (not cleaning up) iterators properly.

To confirm that the `graphql-ws` is working properly, please plug in the [getting started schema](https://github.com/enisdenjo/graphql-ws#create-a-graphql-schema) and try out the client as showcased in the [getting started client](https://github.com/enisdenjo/graphql-ws#use-the-client).

After confirming, my guess is that `withFilter` is the culprit here ([following the abundance of issues around it](https://github.com/apollographql/graphql-subscriptions/issues?q=is%3Aissue+is%3Aopen+withFilter)), please open an issue with [`graphql-subscriptions`](https://github.com/apollographql/graphql-subscriptions) instead.

**Reply by ajmerik121** [reply]
I think you're right , getting started example works as expected , **withFilter** from **graphql-subscriptions** is the real issue here . Thanks for your time.
